src/lib/listeners/ProgressEventListener.ts

```diff
@@ -74,8 +74,8 @@ abstract class ProgressSpinner {
 
 export class RuleSelectionProgressSpinner extends ProgressSpinner implements ProgressEventListener {
 	private isListening: boolean;
-	private engineNames: string[];
-	private completionPercent: number;
+	private engineNames: Set<string>;
```

Comment:
`engineNames` is a `Set` now, so if the different Cores somehow have different engines, all the engines will be featured in the spinner status.